src/graph/expression_graph.h

```diff
@@ -16,9 +16,20 @@
 
 namespace marian {
 
+/**
+ * @brief Create a expression node of any type, and pass all
+ * arguments to any available constructor
+ *
```

Comment:
This is a general comment to the style of Dxygen documentation. What do you think about skipping the empty line between the `@brief` section and the rest of the text to save the vertical space?

It seems that Doxygen supports creating `@brief` sections automatically from the first sentence in the doc. This way we could skip that keyword entirely. Have you maybe tried that? What do you think?

Reply:
If we set JAVADOC_AUTOBRIEF to YES in doxygen configure file, then doxygen will interpret the first line (until the first dot) of a Javadoc-style comment as the brief description. 
I think it can help to remove @brief annotation and reduce extra lines for API docs.